collect_app/src/main/java/org/odk/collect/android/projects/Project.kt

```diff
@@ -1,7 +1,9 @@
 package org.odk.collect.android.projects
 
 data class Project(
-    val uuid: String,
     val name: String,
-    val icon: String
-)
+    val icon: String,
+    val color: String
+) {
+    lateinit var uuid: String
```

Comment:
Oh I really wouldn't recommend using `lateinit` for values like this that may or may not exist. For me, `lateinit` is really only for cases like a JUnit test or `Activity` where part of the object's setup happens outside of the constructor (`@Before` or `onCreate`). Docs on `lateinit` actually reference a test [here](https://kotlinlang.org/docs/properties.html#late-initialized-properties-and-variables).

Here I'd say `uuid` should be a `String?` as it wouldn't be considered exceptional or problematic but makes sure any caller knows they have to check before using it (something we don't get from `lateinit`). 

This can all end up feeling a bit nasty in Kotlin as we might end up with several optional fields on an object (a creation date for instance). A nice way of modelling that if it becomes painful is to do something like this:

```kotlin
abstract class Project {
    abstract val name: String
    abstract val icon: String
    abstract val color: String

    data class New(val name: String, val icon: String, val color: String)
    data class Existing(val uuid: String, val name: String, val icon: String, val color: String)
}
```